Wrong number of arguments: call(p,[1])